src/bin/pg_autoctl/cli_get_set_properties.c

```diff
@@ -0,0 +1,368 @@
+}
+
+
+static void
+cli_get_formation_number_sync_standbys(int argc, char **argv)
+{
+	KeeperConfig config = keeperOptions;
+	Monitor monitor = { 0 };
+	bool missing_pgdata_is_ok = false;
+	bool pg_is_not_running_is_ok = true;
+	int numberSyncStandbys = 0;
+
+	if (!keeper_config_read_file(&config,
+								 missing_pgdata_is_ok,
+								 pg_is_not_running_is_ok))
+	{
+		/* errors have already been logged. */
+		exit(EXIT_CODE_BAD_CONFIG);
+	}
+
+	if (!monitor_init_from_pgsetup(&monitor, &config.pgSetup))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	if (!monitor_get_formation_number_sync_standbys(&monitor,
+			config.formation, &numberSyncStandbys))
+	{
+		exit(EXIT_CODE_MONITOR);
+	}
+
+	fprintf(stdout, "%d\n", numberSyncStandbys);
+}
+
+
+/*
+ * cli_set_node_property sets a node property on the monitor
+ * for to current keeper node
+ */
+static void
+cli_set_node_property(int argc, char **argv)
+{
+	Keeper keeper = { 0 };
+	bool missing_pgdata_is_ok = false;
+	bool pg_is_not_running_is_ok = true;
+	char *name = NULL;
+	char *value = NULL;
+	int intValue = -1;
+	bool boolValue = false;
+	int *candidatePriority = NULL;
+	bool *replicationQuorum = NULL;
+	int assignedCandidatePriority = -1;
+	bool assignedReplicationQuorum = false;
+
+	keeper.config = keeperOptions;
+
+	if (argc != 2)
+	{
+		log_error("Expected node property name and a value, instead found %d arguments", argc);
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	name = argv[0];
+	value = argv[1];
+
+	if (strcmp(name, "candidate-priority") == 0)
+	{
+		intValue = strtol(value, NULL, 10);
+		if (errno == EINVAL || intValue < 0 || intValue > 100)
+		{
+			log_error("candidate-priority value %s is not valid."
+					  " Valid values are integers from 0 to 100. ", value);
+			exit(EXIT_CODE_BAD_ARGS);
+		}
+
+		candidatePriority = &intValue;
+	}
+	else if (strcmp(name, "replication-quorum") == 0)
+	{
+		if (!streq(value, "true") && !streq(value, "false")  )
+		{
+			log_error("replication-quorum value %s is not valid."
+					  " Valid values are true or false", value);
+			exit(EXIT_CODE_BAD_ARGS);
+		}
+
+		boolValue = streq(value, "true");
+		replicationQuorum = &boolValue;
+	}
+	else
+	{
+		log_error("Unknown node property %s", name);
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	if (!keeper_config_read_file(&(keeper.config),
+								 missing_pgdata_is_ok,
+								 pg_is_not_running_is_ok))
+	{
+		/* errors have already been logged. */
+		exit(EXIT_CODE_BAD_CONFIG);
+	}
+
+	if (!keeper_init(&keeper, &keeper.config))
+	{
+		log_fatal("Failed to initialize keeper, see above for details");
+		exit(EXIT_CODE_KEEPER);
+	}
+
+
+	if (!monitor_init(&(keeper.monitor), keeper.config.monitor_pguri))
+	{
+		/* errors have already been logged */
+		exit(EXIT_CODE_BAD_ARGS);
+	}
+
+	if (!monitor_get_node_replication_settings(&(keeper.monitor),
+			keeper.state.current_node_id,
+			&assignedCandidatePriority, &assignedReplicationQuorum))
+	{
+		exit(EXIT_CODE_BAD_ARGS);
+	}
```

Comment:
This technique only works in the happy case. If two commands are run in parallel (by mistake, or because of a bug somewhere in the scripts calling `pg_autoctl`, etc), then we have race conditions all the way. On way to avoid them is to use `UPDATE table SET param = new value WHERE param = old value` but the function call API you've set up on the monitor does not allow that.

I think it would be easier and safer to just have two functions here, one per parameter. Do we have any call-site that wants to set both parameters at the same time?